src/evm/protocol/vm/state_builder.rs

```diff
@@ -425,29 +422,11 @@ impl EVMPoolStateBuilder {
             .simulate(&sim_params)
             .map_err(|err| SimulationError::FatalError(err.to_string()))?;
 
-        let address = decode(&[ParamType::Address], &sim_result.result)
-            .map_err(|_| {
-                SimulationError::FatalError(
-                    "Failed to get address from call: Failed to decode address list from simulation result".into(),
-                )
-            })?
-            .into_iter()
-            .next()
-            .ok_or_else(|| {
-                SimulationError::FatalError(
-                    "Failed to get address from call: Couldn't retrieve address from simulation for stateless contracts".into(),
-                )
-            })?;
+        let address: Address =Address::abi_decode(&sim_result.result, true).map_err(|e| {
```

Comment:
Are you nightly formatting? Spacing looks weird

Reply:
oh yes.. I did 😕 weird